tests/lib/tools/boot-state

```diff
@@ -0,0 +1,203 @@
+#!/bin/bash
+
+show_help() {
+    echo "usage: bootenv show [var]"
+    echo "       bootenv set <var> <value>"
+    echo "       bootenv unset <var>"
+    echo "       boot-path"
+    echo "       wait-core-post-boot"
+    echo ""
+    echo "Get information and manages the boot loader for the current system"
+    echo ""
+    echo "COMMANDS:"
+    echo "  bootenv show:         prints the whole bootenv or just the variable passed as parameter"
+    echo "  bootenv set:          sets the given var and value on boot configuration"
+    echo "  bootenv unset:        unsets the given var from boot configuration"
+    echo "  boot-path:            prints the boot path"
+    echo "  wait-core-post-boot:  waits until the snap_mode bootenv var is empty"
+}
+
+get_grub_editenv() {
+    if [ -e /boot/grub2/grubenv ]; then
+        command -v grub2-editenv
+    elif [ -e /boot/grub/grubenv ]; then
+        command -v grub-editenv
+    fi
+}
+
+run_grub_editenv() {
+    local CMD=$(get_grub_editenv)
+
+    if [ -n "$CMD" ]; then
```

Comment:
Should we just use $(get_grub_editenv) directly instead of having this function? I mean this is simply calling or failing so it will have the same result.

Reply:
sure, lets try that